### ISSUE TITLE
Webpacker 3+ is ignoring 'app/assets' folder 

### ISSUE THREAD

**Issue by HectorLS** [rails/webpacker#1104]
**chunk of package.json**
```
"@rails/webpacker": "^3.0.2",
"webpack-dev-server": "^2.9.5"
```
Hello, this error is so weird because same config in other project was working..
Anyway, i hope you could help me here 😄 

Module build failed: ModuleNotFoundError: Module not found: Error: Can't resolve './assets/fonts/Fontname/Fontname-Semibold.eot' in '/Users/Username/Projectname/app/frontend/packs'

the first path its just fine. but its just looking inside de packs folder.

**chunk of webpacker.yml**
```
# Note: You must restart bin/webpack-dev-server for changes to take effect

default: &default
  source_path: app/frontend
  source_entry_path: packs
  public_output_path: packs
  cache_path: tmp/cache/webpacker

  # Additional paths webpack should lookup modules
  # ['app/assets', 'engine/foo/app/assets']
  resolved_paths: []
```

i tried added  `resolved_paths: ['app/assets']` . but still not working

Yesterday something similar happens with the assets stored in node_modules packages and thanks to @gauravtiwari got fix it this way.

**environment.js**
```
const { environment } = require('@rails/webpacker');
const webpack         = require('webpack');
const path            = require('path');

// Get a pre-configured plugin
environment.plugins.get('ExtractText'); // Is an ExtractTextPlugin instance

// Add an additional plugin of your choosing : ProvidePlugin
environment.plugins.append('Provide',  new webpack.ProvidePlugin({
    $     : 'jquery',
    jQuery: 'jquery',
    jquery: 'jquery'
  })
);

// Take the assets stored in node_modules packages 
const sassLoader      = environment.loaders.get('sass');
const findSassLoader  = loader => (loader === 'sass-loader' || loader.loader === 'sass-loader');
const sassLoaderIndex = sassLoader.use.findIndex(findSassLoader);

sassLoader.use.splice(sassLoaderIndex, 0, {
  loader: 'resolve-url-loader',
  options: {
    attempts: 1
  }
});


// export the updated config
module.exports = environment
```

 So my guess is that should be added something to take asset folder also in count, right 😕  ?

**Comment by gauravtiwari** [comment]
@HectorLS Are you sure paths are correct? 

```js
default: &default
  source_path: app/frontend
  source_entry_path: packs
  public_output_path: packs
  cache_path: tmp/cache/webpacker

  # Additional paths webpack should lookup modules
  # ['app/assets', 'engine/foo/app/assets']
  resolved_paths: []
```

Can't tell more without seeing folder structure.

**Comment by HectorLS** [comment]
you mean that public_output_path: packs should be 'asset/packs' ?

**Comment by gauravtiwari** [comment]
Nope, I meant - if you could paste a screenshot of your directory structure i.e how packs are structured under `app/frontend`

**Comment by gauravtiwari** [comment]
And, resolved path should include `app/assets` if you want to reference assets from that folder. 

```yml
default: &default
  resolved_paths: ['app/assets']
```

**Comment by HectorLS** [comment]
the structure is something like this
```
app/
  - assets /
     -- fonts/
     -- img/
  - frontend/
     -- packs/
         --- scssFile.scss
         --- jsFile.js


```

**Comment by gauravtiwari** [comment]
Where are you using this module? `./assets/fonts/Fontname/Fontname-Semibold.eot`

**Comment by HectorLS** [comment]
I have many imports in my main fail scssFile.scss
app/
  - assets /
     -- fonts/
     -- img/
  - frontend/
     -- packs/
         --- scssFile.scss
         --- jsFile.js
     -- scss/
          --- base/

**Comment by HectorLS** [comment]
Damn it !
the key was '/assets/font' instead of 'assets/fonts'
that first backslash 🤦‍♂️

**Comment by gauravtiwari** [comment]
`resolved_paths` doesn't work for css or scss files AFAIK: https://webpack.js.org/concepts/module-resolution/

The files used in css files are processed using https://github.com/webpack-contrib/file-loader

**Comment by howells** [comment]
@gauravtiwari ah! That'd explain a lot, thanks :-) I'll see what I can find out about how to resolve without cumbersome relative paths

**Comment by onomated** [comment]
@howells Did you figure this out?  This worked pre-3.2.1, and been pulling hairs after updating to 3.4.1

**Comment by howells** [comment]
@onomated afraid not!

**Comment by onomated** [comment]
@gauravtiwari this worked fine before upgrading to 3.4.1.  First had to add `resolve-url-loader` to resolve other module asset paths (wasn't needed before upgrade), and now this is not aware of the `resolved_paths` configuration.  Any advise on what changed?  I would rather patch my local setup back to its prior working configuration, than explicitly resolve paths to what should be a standard `app/assets` path

**Comment by onomated** [comment]
@howells @gauravtiwari  found the changes that limit the previous behavior.
TLDR; comment out the `attempts` option in the `resolve-url-loader` plugin.
As part of the fix for https://github.com/rails/webpacker/issues/932, the `resolve-url-loader` added an `attempts` option that limits the file search scope (?).  The suggestion in https://github.com/rails/webpacker/blob/master/docs/css.md to sets this to 1.  Commenting that out, and falling back to the default value of infinite, resolves files under the `app/assets` directory.  Documentation of the attempts property is a bit murky to me.  Sounds like it should be something between 1 and infinity.  Will read the code to figure that out.
